forc-pkg/src/pkg.rs

```diff
@@ -2369,14 +2399,66 @@ pub fn build_with_options(build_options: BuildOpts) -> Result<Built> {
     };
 
     let manifest_file = ManifestFile::from_dir(&this_dir)?;
-    match manifest_file {
-        ManifestFile::Package(package_manifest) => {
-            let built_package = build_package_with_options(&package_manifest, build_options)?;
-            Ok(Built::Package(Box::new(built_package)))
+    let member_manifests = manifest_file.member_manifests()?;
+    let lock_path = manifest_file.lock_path()?;
+    let build_plan = BuildPlan::from_lock_and_manifests(
+        &lock_path,
+        &member_manifests,
+        build_options.pkg.locked,
+        build_options.pkg.offline,
+    )?;
+    let graph = build_plan.graph();
+    let manifest_map = build_plan.manifest_map();
+    let build_profiles: HashMap<String, BuildProfile> = build_plan.build_profiles().collect();
+    // Get the selected build profile using build options
+    let (profile_name, build_profile) = build_profile_from_opts(&build_profiles, &build_options)?;
+
+    // If this is a workspace we want to have all members in the output.
+    let outputs = match &manifest_file {
+        ManifestFile::Package(pkg_manifest) => std::iter::once(
+            build_plan
+                .find_member_index(&pkg_manifest.project.name)
+                .ok_or_else(|| anyhow!("Cannot found project node in the graph"))?,
+        )
+        .collect(),
+        ManifestFile::Workspace(_) => build_plan.member_nodes().collect(),
+    };
```

Comment:
Basically this part substitutes for `node_plan` approach in the case of building just a single pkg of a workspace. Since any member of the workspace would result all of the member nodes, if we just want to build one of them I ended up checking like this. Otherwise it ends up building all of the members while building any member of the workspace individually.

Reply:
Nice one, makes sense and looks good!